src/domain/useCases/fileTree/__tests__/ToggleIsReward.test.ts

```diff
@@ -0,0 +1,42 @@
+import 'jest';
+import 'reflect-metadata';
+import {SchemaFile} from '../../../states/objects/fileTree/SchemaFile';
+import {SchemaTree} from '../../../states/objects/fileTree/SchemaTree';
+import {Project} from '../../../states/objects/Project';
+import {ProjectConfig} from '../../../states/objects/ProjectConfig';
+import {ProjectState} from '../../../states/ProjectState';
+import {ToggleIsRewardFile} from "../ToggleIsRewardFile";
+
+let useCase: ToggleIsRewardFile;
+let projectState: ProjectState;
+
+beforeEach(() => {
+    projectState = new ProjectState();
+
+    let fileA: SchemaFile = new SchemaFile('File A', 'normalFile', 'file', 'file');
+    let fileB: SchemaFile = new SchemaFile('File B', 'rewardFile', 'file', 'file');
+    fileB.setIsReward(true);
+
+    const tree = new SchemaTree([fileA, fileB]);
+    projectState.setLoaded(new Project('/root/projectFile', new ProjectConfig({name: ''}), tree));
+
+    useCase = new ToggleIsRewardFile(projectState);
+});
+
+describe('ToggleIsReward', () => {
```

Comment:
Why are always both files effected in the test cases?

Isn't the logic build in a way that it checks the base name of a file? So when we eg. execute `useCase.execute('normalFile')`, shouldn't only `fileA` be affected?

Reply:
They're not both affected.
In L. 17 I set fileB to be a reward in the beforeEach().
So if I now toggle fileA to also be a reward for the first test case, both are true.
If I now toggle fileB to no longer be a reward for the second test case, both are false.

Reply:
Ah, then I was just too stupid to read the test 😄